rotkehlchen/assets/utils.py

```diff
@@ -73,10 +74,21 @@ def get_or_create_evm_token(
             form_with_incomplete_data=form_with_incomplete_data,
         )
     except (UnknownAsset, DeserializationError):
+        # It can happen that the assets exists but is missing basic information. Check if it exists
```

Comment:
again when can that happen? Can someone explain this to me?

Reply:
One thing that can happen is that we try to recover a token that we failed to add due to a remote error or that that was not correctly filled with the user.

In the previous logic what would happen is:

1. Get the asset with `form_with_incomplete_data=False`
2. Fail since for example the decimals field is missing
3. Fetch the information
4. Try to add the token
5. Fail because the token already exists

What this PR does is:

1. If we fail to get the information check the queried properties
2. If any of the queried properties is missing fail
3. if we got the properties but any is missing update the asset
4. if the asset was not missing in the db and we got the properties add it